plan/predicate_push_down.go

```diff
@@ -47,10 +48,225 @@ func columnSubstitute(expr expression.Expression, schema expression.Schema, newE
 	return expr
 }
 
+// propagateConstant propagate constant values of equality predicates and inequality predicates in a condition.
+func propagateConstant(conditions []expression.Expression) []expression.Expression {
```

Comment:
define arguments as `conditions ...expression.Expression`, so you can pass in one expression.